official/resnet/imagenet_main.py

```diff
@@ -291,5 +291,5 @@ def main(unused_argv):
 
   parser = resnet_run_loop.ResnetArgParser(
       resnet_size_choices=[18, 34, 50, 101, 152, 200])
-  FLAGS, unparsed = parser.parse_known_args()
-  tf.app.run(argv=[sys.argv[0]] + unparsed)
+  FLAGS = parser.parse_args()
+  tf.app.run(argv=[sys.argv[0]])
```

Comment:
I think my question is that why the parser need to be in __main__ instead of main() function.

The code in __main__ is only executed when imagenet_main.py is invoked as main target instead of imported as module. For this file, I can hardly foresee it being import as a module for real prod doc, except the case for test. If this assumption is true, then why not push everything here to main(), so that your test code and just import it as module, and avoid calling a new python process.

Reply:
Discussed offline and agreed that this is the way to go.